version: "3.9"

name: <%= environment_name %>

services:
  postgres:
    container_name: <%= postgres_container_name %>
    image: postgres:15.3-alpine
    command: postgres -c 'max_connections=300'
    hostname: postgres-15-3    
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      interval: 10s
      timeout: 5s
      retries: 5
    volumes:
       - "pgdata:/var/lib/postgresql/data"
       - ./db:/db
    ports:
      - "5432:5432"
    networks:
      - service_network
    restart: unless-stopped
  
  pgadmin:
    container_name: <%= pgadmin_container_name %>
    image: dpage/pgadmin4:7.7
    hostname: pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: pgadmin4@pgadmin.org
      PGADMIN_DEFAULT_PASSWORD: admin
    depends_on:
      postgres:
        condition: service_healthy
    volumes:
       - "pgadmin:/var/lib/pgadmin"
    ports:
      - "8080:80"
      - "8443:443"
    networks:
      - service_network
    restart: unless-stopped

  redis:
    container_name: <%= redis_container_name %>
    image: redis:7.0-alpine
    hostname: redis-7
    restart: always
    ports:
      - "6379:6379"
    networks:
      - service_network
    command: redis-server --requirepass redispwd
    volumes: 
      - redis:/data

networks:
  service_network:
    name: <%= network_name %>
    external: true

volumes:
  pgdata:
    driver: local
  pgadmin:
    driver: local
  redis:
    driver: local
 